administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,46 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		$app    = JFactory::getApplication();
+		
+		if (!JSession::checkToken())
+		{
+			$app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			echo new JResponseJson;
```

Comment:
The response needs an error status set. Now it has no error status as if everything is OK.